src/storage.ts

```diff
@@ -0,0 +1,56 @@
+import { v4 as uuid } from 'uuid';
+
+export class ItemsStorage<T extends { id: string; isDeleted: boolean }> {
```

Comment:
Just a thing for you to think about. `ItemsStorage` looks like an implementation of a Repository pattern, which basically a facade for the business domain models that hides some implementation details. Since `ItemsStorage` is aware of `isDeleted` property it makes sense to encapsulate operations with `isDeleted` property (i.e. query only those items with `isDeleted = false`). Thus we don't have to worry about `isDeleted` value on more abstract levels (e.g. in `getAutoSuggestUsers`)

Btw, :+1:  for making such abstraction :slightly_smiling_face: